src/apps/api/serializers/queues.py

```diff
@@ -30,28 +32,33 @@ class Meta:
             'created_when',
             'is_owner',
             'id',
-        ]
-        read_only_fields = [
+        )
+        read_only_fields = (
             'broker_url',
             'vhost',
             'created_when',
             'is_owner',
-        ]
+        )
+
+    def validate(self, attrs):
+        request = self.context.get('request')
+        if request.user.queues.count() == request.user.rabbitmq_queue_limit and not request.user.is_superuser:
+            # TODO: Do we want to raise a validation error (400, bad request) or something that will raise a 403?
+            raise ValidationError("User has reached queue limit!")
+        return super().validate(attrs)
 
     def get_is_owner(self, instance):
-        if instance.owner == self.context.get('owner'):
-            return True
-        return False
+        return instance.owner == self.context.get('owner')
```

Comment:
would it be possible for a queue to have no owner, and `context.get('owner')` to also be empty? in which case we would be saying: `return None == None` i.e. `is_owner=True` which we obviously don't wanna do?
Maybe do `return instance.owner and instance.owner == self.context.get('owner')`